mne/epochs.py

```diff
@@ -662,7 +662,7 @@ def __next__(self, *args, **kwargs):
     def __hash__(self):
         if not self.preload:
             raise RuntimeError('Cannot hash epochs unless preloaded')
-        return object_hash(dict(info=self.info, data=self._data))
+        return object_hash(dict(info=hash(self.info), data=self._data))
```

Comment:
It occurred to me that it will differ (at least) because `_store` is missing two fields, `ch_names` and `nchan` that used to be there.

It also occured to me that it might be acceptable for the hashes to differ so long as we document it properly -- i.e. the pain of people having to re-run some bits of analysis might be acceptable to get this PR in. @dengemann do you expect a change in hashing functions to break many of your analysis pipelines? I assume you've been using that feature most heavily, if at all. @jasmainak perhaps you have, too, since it's probably used by joblib caching sometimes?